docs/operators/replace.md

```diff
@@ -0,0 +1,87 @@
+---
+title: replace
+category: Modify
+example: 'replace what=42, with=null'
+---
+
+Replaces all occurrences of a value with another value.
+
+```tql
+replace [path:field...], what=any, with=any
+```
+
+## Description
+
+The `replace` operator scans all fields of each input event and replaces every
+occurrence of a value equal to `what` with the value specified by `with`.
+
+### `path: field... (optional)`
+
+An optional set of paths to restrict replacements in.
+
+### `what: any`
+
+The value to search for and replace.
+
+### `with: any`
+
+The value to replace in place of `what`.
+
+## Examples
+
+### Replace all occurrences of 42 with null
+
+```tql
+from {
+  count: 42,
+  data: {value: 42, other: 100},
+  list: [42, 24, 42]
+}
+replace what=42, with=null
+```
```

Comment:
For me, the names "what" and "with" do not read well. It sounds very unnatural.

Did you consider alternative names such as `replace old=42, new=null` or `replace find=42, value=null`? Or maybe `replace 42, null`?

Reply:
@raxyte Can you drive the conversation forward on how the interface should look like? We still need to address this before we can merge.

Reply:
I and @IyeOnline are fine with the current design, the other alternatives either provide no value, or are worse (like positional replacement options, as they do not make it clear what is to be replaced with what).